src/PKSim.Core/Snapshots/Mappers/ProjectMapper.cs

```diff
@@ -35,11 +39,56 @@ public ProjectMapper(IExecutionContext executionContext, SimulationMapper simula
          snapshot.Formulations = await mapBuildingBlocksToSnapshots<Formulation>(project.All<Model.Formulation>());
          snapshot.Protocols = await mapBuildingBlocksToSnapshots<Protocol>(project.All<Model.Protocol>());
          snapshot.Populations = await mapBuildingBlocksToSnapshots<Population>(project.All<Model.Population>());
-         snapshot.ObservedData = await mapObservedDataToSnapshots(project.AllObservedData);
          snapshot.Simulations = await mapSimulationsToSnapshots(project.All<Model.Simulation>(), project);
+         snapshot.ObservedData = await mapObservedDataToSnapshots(project.AllObservedData);
+         snapshot.ObservedDataClassifications = await mapObservedDataClassificationsToSnapshots(
+            observedDataClassifiablesFrom<ModelObservedDataClassification>(project.AllClassifications),
+            observedDataClassifiablesFrom<ModelObservedDataClassifiable>(project.AllClassifiables));
+
          return snapshot;
       }
 
+      private Task<SnapshotDataRepository[]> mapObservedDataToSnapshots(IReadOnlyCollection<ModelDataRepository> allObservedData)
+      {
+         return mapModelsToSnapshot<ModelDataRepository, SnapshotDataRepository>(allObservedData, snapshotMapper.MapToSnapshot);
+      }
+
+      private void addObservedDataToProject(ModelProject project, ModelDataRepository repository)
+      {
+         project.AddObservedData(repository);
+         project.GetOrCreateClassifiableFor<ModelObservedDataClassifiable, ModelDataRepository>(repository);
+      }
+
+      private Task<IEnumerable<ModelDataRepository>> observedDataFrom(SnapshotDataRepository[] snapshotRepositories)
+      {
+         return awaitAs<ModelDataRepository>(mapSnapshotsToModels(snapshotRepositories));
+      }
+
+      private IEnumerable<T> findRoots<T>(IEnumerable<T> classifications) where T : IClassifiable
+      {
+         return classifications.Where(x => x.Parent == null);
+      }
+
+      private IReadOnlyList<T> observedDataClassifiablesFrom<T>(IEnumerable<IClassifiable> classifications) where T : IClassifiable
+      {
+         return classifications.Where(x => x.ClassificationType == ClassificationType.ObservedData).OfType<T>().ToList();
+      }
+
+      private Task<Classification[]> mapObservedDataClassificationsToSnapshots(IReadOnlyList<ModelObservedDataClassification> allClassifications, IReadOnlyList<ModelObservedDataClassifiable> allClassifiables)
+      {
+         var observedDataClassificationsContext = new ClassificationContext
+         {
+            Classifications = allClassifications,
+            Classifiables = allClassifiables.Select(x => new ClassifiableContext {Name = x.Name, Parent = x.Parent}).ToList()
```

Comment:
Why the ClassifiableContext? Can't you use the classifiable? (Name, parent are both properties of the classifiable already)

Reply:
Requires Classifiable<T>. There's no non-generic at the moment.

Reply:
Classifiable<T>

Reply:
Generic is removed from my comments above
```
Classifiable<T>
```

Reply:
isn't IClassifiableWrapper what you need?